thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/validators/DetectionConfigValidator.java

```diff
@@ -72,20 +72,15 @@ public DetectionConfigValidator(DataProvider provider) {
    * Validate the pipeline by loading and initializing components
    */
   private void semanticValidation(DetectionConfigDTO detectionConfig) {
-    try {
-      // backup and swap out id
-      Long id = detectionConfig.getId();
-      detectionConfig.setId(-1L);
-
-      // try to load the detection pipeline and init all the components
-      this.loader.from(provider, detectionConfig, 0, 0);
-
-      // set id back
-      detectionConfig.setId(id);
-    } catch (Exception e){
-      // exception thrown in validate pipeline via reflection
-      throw new IllegalArgumentException("Semantic error: " + e.getCause().getMessage());
```

Comment:
will delete this swallow the error message?

Reply:
We would catch re-throw only if we add some additional information that can help us debug. But there we are simply changing the exception type without enriching the message.

Without the catch, the exception will be propagated up and handled in YamlResource.

Reply:
make sense.